styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java

```diff
@@ -20,168 +20,24 @@
-      e.add("workflow id cannot be empty");
-    }
-
-    if (!workflowId.id().equals(cfg.id())) {
-      e.add("workflow id mismatch");
-    }
-
-    // TODO: validate more of the contents
-
-    upperLimit(e, cfg.id().length(),
-        MAX_ID_LENGTH, "id too long");
-    upperLimit(e, cfg.commitSha().map(String::length).orElse(0),
-        MAX_COMMIT_SHA_LENGTH, "commitSha too long");
-    upperLimit(e, cfg.secret().map(s -> s.name().length()).orElse(0),
-        MAX_SECRET_NAME_LENGTH, "secret name too long");
-    upperLimit(e, cfg.secret().map(s -> s.mountPath().length()).orElse(0),
-        MAX_SECRET_MOUNT_PATH_LENGTH, "secret mount path too long");
-    upperLimit(e, cfg.serviceAccount().map(String::length).orElse(0),
-        MAX_SERVICE_ACCOUNT_LENGTH, "service account too long");
-    upperLimit(e, cfg.resources().size(),
-        MAX_RESOURCES, "too many resources");
-    upperLimit(e, cfg.env().size(),
-        MAX_ENV_VARS, "too many env vars");
-    upperLimit(e, cfg.env().entrySet().stream()
-            .mapToInt(entry -> entry.getKey().length() + entry.getValue().length()).sum(),
-        MAX_ENV_SIZE, "env too big");
 
-    cfg.dockerImage().ifPresent(image ->
-        dockerImageValidator.validateImageReference(image).stream()
-            .map(s -> "invalid image: " + s)
-            .forEach(e::add));
+public interface WorkflowValidator {
 
-    cfg.resources().stream().map(String::length).forEach(v ->
-        upperLimit(e, v, MAX_RESOURCE_LENGTH, "resource name too long"));
+  List<String> validateWorkflow(Workflow workflow);
 
-    cfg.dockerArgs().ifPresent(args -> {
-      final int dockerArgs = args.size() + args.stream().mapToInt(String::length).sum();
-      upperLimit(e, dockerArgs, MAX_DOCKER_ARGS_TOTAL, "docker args is too large");
-    });
-
-    cfg.offset().ifPresent(offset -> {
-      try {
-        TimeUtil.addOffset(ZonedDateTime.now(), offset);
-      } catch (DateTimeParseException ex) {
-        e.add(format("invalid offset: %s", ex.getMessage()));
-      }
-    });
-
-    try {
-      TimeUtil.cron(cfg.schedule());
-    } catch (IllegalArgumentException ex) {
-      e.add("invalid schedule");
-    }
-
-    cfg.runningTimeout().ifPresent(timeout -> {
-      lowerLimit(e, timeout, MIN_RUNNING_TIMEOUT, "running timeout is too small");
-      if (maybeMaxRunningTimeout != null) {
-        upperLimit(e, timeout, maybeMaxRunningTimeout, "running timeout is too big");
-      }
-    });
-
-    cfg.secret().ifPresent(secret -> {
-      if (secretWhitelist != null && !secretWhitelist.contains(secret.name())) {
-        e.add("secret " + secret.name() + " is not whitelisted");
-      }
-    });
-
-    return e;
-  }
-
-  private <T extends Comparable<T>> void lowerLimit(List<String> errors, T value, T limit, String message) {
-    limit(errors,value.compareTo(limit) < 0, value, limit, message);
+  default <T extends Comparable<T>> void lowerLimit(List<String> errors, T value, T limit, String message) {
```

Comment:
I see this methods as auxiliary methods and not part of the interface. So it would be better to have it as statics methods instead of default methods